packages/offix-server-conflicts/src/states/VersionedObjectState.ts

```diff
@@ -0,0 +1,53 @@
+import { ObjectState } from "../api/ObjectState";
+import { ObjectStateData } from "../api/ObjectStateData";
+import { ObjectConflictError } from "../api/ObjectConflictError";
+
+/**
+ * Object state manager using a version field
+ * Detects conflicts and allows moving to next state using the version field of the object
+ *
+ * VersionedObjectState requires GraphQL types to contain version field.
+ * For example:
+ *
+ * type User {
+ *   id: ID!
+ *   version: String
+ * }
+ */
+export class VersionedObjectState implements ObjectState {
+
+  public checkForConflict(serverState: ObjectStateData, clientState: ObjectStateData) {
+    if (serverState.version && clientState.version) {
+      if (serverState.version !== clientState.version) {
+        const filteredServerState: any = {};
+        for (const field in clientState) {
```

Comment:
Minor nit but I think this method of looping of an object's own properties is kinda strange. I think that this is more readable

```
for (const key in Object.keys(clientState)) {
 // filteredServerState[field] = serverState[field];
}
```

No big deal either way though :)

Reply:
Good advice. we need the same on client